catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JsonPatchMetadataOperationMapper.java

```diff
@@ -0,0 +1,38 @@
+package org.openmetadata.catalog.security;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.json.JsonPatch;
+import org.openmetadata.catalog.type.MetadataOperation;
+import org.openmetadata.catalog.util.JsonUtils;
+
+public class JsonPatchMetadataOperationMapper {
+
+  public static List<MetadataOperation> getMetadataOperations(JsonPatch jsonPatch) {
+    return jsonPatch.toJsonArray().stream()
+        .map(JsonPatchMetadataOperationMapper::getMetadataOperation)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  private static MetadataOperation getMetadataOperation(Object jsonPatchObject) {
+    Map<String, Object> jsonPatchMap = JsonUtils.getMap(jsonPatchObject);
+    String path = jsonPatchMap.get("path").toString();
+
+    // To get operation, use the following:
+    // JsonPatch.Operation op =  JsonPatch.Operation.fromOperationName(jsonPatchMap.get("op").toString());
+
+    if (path.contains("description")) {
```

Comment:
What other operations should we cover?

Reply:
1. follow/un-follow , by default this should be allowed
2. Sample-data can be guarded as well (later versions)
3. Tests etc..(future)

Reply:
#2228